plugins/image-prioritizer/helper.php

```diff
@@ -10,6 +10,41 @@
 	exit; // Exit if accessed directly.
 }
 
+/**
+ * Initializes Image Prioritizer when Optimization Detective has loaded.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $optimization_detective_version Current version of the optimization detective plugin.
+ */
+function image_prioritizer_init( string $optimization_detective_version ): void {
+	$required_od_version = '0.7.0';
+	if ( ! version_compare( (string) strtok( $optimization_detective_version, '-' ), $required_od_version, '>=' ) ) {
+		add_action(
+			'admin_notices',
+			static function (): void {
+				global $pagenow;
+				if ( ! in_array( $pagenow, array( 'index.php', 'plugins.php' ), true ) ) {
+					return;
+				}
+				wp_admin_notice(
+					esc_html__( 'The Image Prioritizer plugin requires a newer version of the Optimization Detective plugin. Please update your plugins.', 'image-prioritizer' ),
```

Comment:
can we add a link to the plugin update screen here?

Reply:
We could, but the user may not have the capability to access the plugin update screen. Plugins may not be updatable on the environment either due to it being a read only filesystem. So these complications would seem to make adding a link maybe more trouble than its worth.

Reply:
ah, good point. I was trying to make this more helpful... maybe don't show the message to users that can't upgrade plugins?

Reply:
True, but I think this might be overkill. It would be unlikely that someone would update Image Prioritizer but not also update Optimization Detective.